cedar-policy-core/src/ast/policy_set.rs

```diff
@@ -187,6 +228,31 @@ impl PolicySet {
         }
     }
 
+    /// Remove a template from the policy set.
+    /// This involves an expensive scan over all linked policies to see if they are linked to the template
```

Comment:
Not sure if this is what we should finalize on, but it seems reasonable.

Reply:
@aaronjeline Not sure if this is what we want.

Pros: If don't allow you to remove templates with active links, people can't be confused about what will happen.

Cons: If `T1` has 0 active links and `T2` has 10,000 and you want to remove `T1`, you still need to look at 10,000 links.

Reply:
Yeah, what we really want is a two way map. :(